src/main/Application.h

```diff
@@ -278,15 +278,20 @@ class Application
 
     virtual AbstractLedgerTxnParent& getLedgerTxnRoot() = 0;
 
+    // For tests to perform operations on databases while they're
+    // still using old schemas.
+    virtual void actBeforeDBSchemaUpgrade() = 0;
```

Comment:
I think it's the comment that should change -- there's nothing about it that specifically requires the client to be a test; it could be any client of Application.  I'll do that.